[FIX] account_check: Multiple fixes

zaoral opened the pull request.

zaoral committed:
[FIX] account_check: Reconcile rejected checks

In 11.0 version only move lines with payments are take into account in the
reconcile wizard. In order to this now when we reject a check we create an
account payment instead of a account.move. Now ww add/use a new
account_payment_method named "Rejected Third Check"

zaoral committed:
[FIX] account_check: Be able to properly delete rejected check operations

zaoral committed:
[FIX] account_check: Cancel conciliation of bank statement line now is
deleting the related check operation.

jjscarafia commented on account_check/models/account_payment.py:
```diff
@@ -544,3 +554,13 @@ def print_checks(self):
+        rejected_check_account = self.company_id._get_check_account('rejected')
+        for rec in self.filtered(
+                lambda x: x.payment_method_id == reject_check_method):
+            rec.destination_account_id = rejected_check_account.id,
```
remove "," at the end of the line

zaoral committed:
fixup! [FIX] account_check: Reconcile rejected checks

zaoral commented:
@jjscarafia fixed suggested change (I do not know why but I do not find your original comment)

zaoral committed:
[ADD] account_check: update module version

zaoral committed:
[IMP] account_check: Do not use partner_type for check rejections

Do not use post() because it use sequence_code we do not have. Instead we
do the same inside the post() method without the sequence ( create move, and
then set payment to posted)

zaoral committed:
[ADD] account_check: New version without account payment methods. Also work
debit.
